### ISSUE TITLE
Datetime Widget with format option results in "Invalid date" when user enters empty string

### ISSUE THREAD

**Issue by SwissWoodSolutions** [netlify/netlify-cms#2697]
**Describe the bug**

Let's say you have the following in your netlify-cms `config.yml`:

    fields:
      - {label: "Datum", name: "date", widget: "datetime", format: "YYYY-MM-DD HH:mm", required: false, default: ""}

The idea is to have a non-required timezone-less datetime.

When the netlify-cms user deletes the field (enters empty string), then

**Actual behaviour**: the string  "Invalid date" is written to the YAML file

**Expected behaviour**: the empty string is written to the YAML

Seems like the [isValidDate validation](https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-widget-datetime/src/DateTimeControl.js#L58) happens before `.format` is applied, and unfortunately:

     moment('').format("YYYY-MM-DD HH:mm") === "Invalid date"

I suppose, changing

```
      const formattedValue = moment(datetime).format(format);
      onChange(formattedValue);
```

to

```
      const formattedValue = datetime === "" ? "" : moment(datetime).format(format);
      onChange(formattedValue);
```

would fix this.

**Applicable Versions:**

 - Netlify CMS version: 2.9.7

**Comment by erquhart** [comment]
Good catch, would take a fix for this. Simpler and less specific would be:

```
const formattedValue = datetime ? moment(datetime).format(format) : '';
onChange(formattedValue);
```

**Comment by mb21** [comment]
Sure, I made a pull request with my normal github account (I was using the SwissWoodSolutions for a netlify website project).

**Comment by Bartunek** [comment]
Any chance the fix from #2705 will be merged? 🙏  I'm fighting with this in client project and it's kinda annoying to add checking if dates are equal to `Invalid date` string.

**Comment by Bartunek** [comment]
@erquhart Wow, that was fast! 😲 Thank you! And of course thanks to @mb21 👍 👏
